instrumentation/influxdb-2.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/influxdb/v2_4/InfluxDbImplInstrumentation.java

```diff
@@ -70,89 +70,71 @@ public void transform(TypeTransformer transformer) {
+    public static Object[] onEnter(
+        @Advice.AllArguments(typing = Assigner.Typing.DYNAMIC) Object[] arguments,
+        @Advice.FieldValue(value = "retrofit") Retrofit retrofit) {
+      CallDepth callDepth = CallDepth.forClass(InfluxDBImpl.class);
       if (callDepth.getAndIncrement() > 0) {
-        return;
+        return new Object[] {arguments};
       }
 
+      Query query = arguments[0] instanceof Query ? (Query) arguments[0] : null;
       if (query == null) {
-        return;
+        return new Object[] {arguments};
       }
       Context parentContext = currentContext();
 
       HttpUrl httpUrl = retrofit.baseUrl();
-      influxDbRequest =
+      InfluxDbRequest influxDbRequest =
           InfluxDbRequest.create(
               httpUrl.host(), httpUrl.port(), query.getDatabase(), null, query.getCommand());
 
-      if (!instrumenter().shouldStart(parentContext, influxDbRequest)) {
-        return;
+      Instrumenter<InfluxDbRequest, Void> instrumenter = instrumenter();
+      if (!instrumenter.shouldStart(parentContext, influxDbRequest)) {
+        return new Object[] {arguments};
       }
 
       // wrap callbacks so they'd run in the context of the parent span
       Object[] newArguments = new Object[arguments.length];
-      boolean hasChangedArgument = false;
       for (int i = 0; i < arguments.length; i++) {
         newArguments[i] = InfluxDbObjetWrapper.wrap(arguments[i], parentContext);
-        hasChangedArgument |= newArguments[i] != arguments[i];
-      }
-      if (hasChangedArgument) {
-        arguments = newArguments;
       }
 
-      context = instrumenter().start(parentContext, influxDbRequest);
-      scope = context.makeCurrent();
+      return new Object[] {
+        newArguments, InfluxDbScope.start(instrumenter, parentContext, influxDbRequest)
+      };
     }
 
     @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
     public static void onExit(
-        @Advice.Thrown Throwable throwable,
-        @Advice.Local("otelCallDepth") CallDepth callDepth,
-        @Advice.Local("otelRequest") InfluxDbRequest influxDbRequest,
-        @Advice.Local("otelContext") Context context,
-        @Advice.Local("otelScope") Scope scope) {
-
+        @Advice.Thrown Throwable throwable, @Advice.Enter Object[] enterArgs) {
+      CallDepth callDepth = CallDepth.forClass(InfluxDBImpl.class);
       if (callDepth.decrementAndGet() > 0) {
         return;
       }
 
-      if (scope == null) {
-        return;
-      }
+      Object extraObject = enterArgs[enterArgs.length - 1];
 
-      scope.close();
-
-      instrumenter().end(context, influxDbRequest, null, throwable);
+      InfluxDbScope.end(extraObject, instrumenter(), throwable);
     }
   }
 
   @SuppressWarnings("unused")
   public static class InfluxDbModifyAdvice {
 
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void onEnter(
+    public static Object onEnter(
```

Comment:
I think now we could return `InfluxDbScope` instead of `Object`

Reply:
Yes, I just made the change and it's working fine.